fdbserver/EncryptKeyProxy.actor.cpp

```diff
@@ -43,43 +43,95 @@
 #include "flow/network.h"
 
 #include <boost/mpl/not.hpp>
+#include <limits>
 #include <string>
 #include <utility>
 #include <memory>
 
 #include "flow/actorcompiler.h" // This must be the last #include.
 
 namespace {
+
+struct CipherKeyValidityTS {
+	int64_t refreshAtTS;
+	int64_t expAtTS;
+};
+
 bool canReplyWith(Error e) {
 	switch (e.code()) {
 	case error_code_encrypt_key_not_found:
+	// FDB <-> KMS connection may be observing transient issues
+	// Caller processes should consider reusing 'non-revocable' CipherKeys iff ONLY below error codes lead to CipherKey
+	// refresh failure
+	case error_code_timed_out:
+	case error_code_connection_failed:
 		return true;
 	default:
 		return false;
 	}
 }
 
+CipherKeyValidityTS getCipherKeyValidityTS(Optional<int64_t> refreshInterval, Optional<int64_t> expiryInterval) {
+	int64_t currTS = (int64_t)now();
```

Comment:
time units in our codebase is usually of double type. not sure if we want to be consistent.

Reply:
Yeah and in some ways `double` seems like an interesting choice :)
Code uses `-1` => `never expire` so I wanted to use `int64_t` which would be sufficient to represent `time` :)

Reply:
sure. I too feeling the double `time` type is sometimes convenient while other times not so.